apis/cloudscale/v1/bucket_types.go

```diff
@@ -36,10 +36,8 @@ type BucketParameters struct {
 
 	// +kubebuilder:validation:Required
 
-	// EndpointURL is the URL where to create the bucket.
-	// If the scheme is omitted (`http/s`), HTTPS is assumed.
-	// Changing the endpoint after initial creation might have no effect.
-	EndpointURL string `json:"endpointURL"`
+	// Deprecated: Only here for compatibility with legacy Bucket objects
+	EndpointURL string `json:"endpointURL,omitempty"`
```

Comment:
We might actually be able to delete this field (to be tested though). AFAIK Kubernetes automatically removes unknown fields from etcd if the field doesn't exist in the scheme anymore.
But probably we should make sure that compositions don't try to set this field anymore, otherwise reconciliations might become stuck as long as the compositions aren't updated.

Reply:
I was thinking more along the lines of:

Deploy everything. Manually remove the forProvider.endpointURL field from all kubernetes objects that have it. Then remove the field from the code. But :shrug:

Reply:
yeah that works too. let's keep it for now, doesn't hurt.